### PR TITLE
[Bugfix] Fix maven version in bamboo Dockerfile

### PR DESCRIPTION
**Pull request by lschlesinger**
### Checklist
Regarding testing, I can only confirm that the command `docker-compose -f src/main/docker/atlassian.yml up -d` for my local dev setup has only worked when I changed the maven version in the respective Dockerfile for bamboo.

### Motivation and Context

The maven version that is set in the bamboo Dockerfile should be changed.
As pointed out by @florian-glombik we should also update the java version to 16 in the course of this PR.

### Description

I recently did the local atlassian setup and ran into a problem with the bamboo Dockerfile used when running the command: `docker-compose -f src/main/docker/atlassian.yml up -d`  as described in the [documentation](https://github.com/ls1intum/Artemis/blob/a5ef79a6c1b13ae1714027928fc738b546144d47/docs/dev/setup/bamboo-bitbucket-jira.rst):

```shell
#5 5.080 E: Version '3.6.0*' for 'maven' was not found ------ executor failed running [/bin/sh -c apt-get update && apt-get install -y --no-install-recommends software-properties-common apt-utils maven=3.6.0*]: exit code: 100 ERROR: Service 'bamboo' failed to build
```

This problem was also observed by @florian-glombik.

To solve it, I changed the specified maven version to `3.6.*` instead of `3.6.0*` the respective bamboo Dockerfile and it worked for me. 

### Steps for Testing
After pulling the proposed changes, the command `docker-compose -f src/main/docker/atlassian.yml up -d` should work without errors. It would be good, if you can confirm that.

### CODE REVIEW

File: src/main/docker/bamboo/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ ENV DEBIAN_FRONTEND noninteractive
 RUN apt-get update && apt-get install -y --no-install-recommends \
     software-properties-common \
     apt-utils \
-    maven=3.6.0*
+    maven=3.6.*
 RUN add-apt-repository -y ppa:linuxuprising/java
 RUN echo debconf shared/accepted-oracle-license-v1-2 select true | debconf-set-selections && echo debconf shared/accepted-oracle-license-v1-2 seen true | debconf-set-selections
 RUN apt-get install -y oracle-java15-installer
```
Comment (florian-glombik) [inline src/main/docker/bamboo/Dockerfile:14]: 15 -> 16 in line 18 aswell

```suggestion
RUN apt-get install -y oracle-java16-installer
```

File: src/main/docker/bamboo/swift/Dockerfile [hunk 2]
```diff
@@ -8,7 +8,7 @@ ENV DEBIAN_FRONTEND noninteractive
 RUN apt-get update && apt-get install -y --no-install-recommends \
     software-properties-common \
     apt-utils \
-    maven=3.6.0*
+    maven=3.6.*
 RUN add-apt-repository -y ppa:linuxuprising/java
 RUN echo debconf shared/accepted-oracle-license-v1-2 select true | debconf-set-selections && echo debconf shared/accepted-oracle-license-v1-2 seen true | debconf-set-selections
 RUN apt-get install -y oracle-java15-installer
```
Comment (florian-glombik) [inline src/main/docker/bamboo/swift/Dockerfile:14]: 15 -> 16 in line 18 aswell

```suggestion
RUN apt-get install -y oracle-java16-installer
```

**Review by florian-glombik — CHANGES_REQUESTED** [review]
Since we are already touching the Docker files I think we should also update the java version from 15 to 16

**Comment by lschlesinger** [comment]
> Since we are already touching the Docker files I think the java version should be updated from 15 to 16

Yes you are totally right! I've accepted your suggestions directly.

**Review by florian-glombik — APPROVED** [review]
Tested it on my windows 10 machine, the `src/main/docker/bamboo/Dockerfile` works fine 👍 

The swift version `src/main/docker/bamboo/swift/Dockerfile` does not work for me. This does not seem to be an issue with the changes made, so it should not stop the PR from being merged. However, if you can fix that / the problem doesn't occur for you that would be even better. 

 Line 21 in the swift Dockerfile seems to be broken for me:
<details><pre>
   => ERROR [stage-0  9/11] COPY --from=swift:latest /usr /usr              34.3s
------
 > [stage-0  9/11] COPY --from=swift:latest /usr /usr:
------
cannot replace to directory /var/lib/docker/overlay2/rka33oaqkso2lk20n1wusuudk/merged/usr/share/doc/perl-base with file
ERROR: Service 'bamboo' failed to build
</pre></details>

**Review by krusche — APPROVED** [review]
Thanks for fixing these issues 👍

**Pull request merged**